### PR TITLE
Adding a Constructor

### PR DESCRIPTION
**Pull request by kraenhansen**
This is an attempt at fixing https://github.com/sgodwincs/webidl-rs/issues/22

This is work in progress, since the added `parse_constructor_operation` test doesn't pass.
I would appreciate another set of eyes on the suggested change.

### CODE REVIEW

**Comment by sgodwincs** [comment]
Just a quick pass, but I think you'll need to add the keyword "constructor" to the lexer here: https://github.com/sgodwincs/webidl-rs/blob/master/src/lexer/mod.rs#L574

**Comment by kraenhansen** [comment]
That did the trick - thanks!

File: src/lexer/mod.rs [hunk 1]
```diff
@@ -610,11 +611,13 @@ impl<'input> Lexer<'input> {
...
-            } else {
-                Token::Identifier(identifier)
-            },
+            _ => {
+                if identifier.starts_with('_') {
+                    Token::Identifier(identifier.split_at(1).1.to_string())
+                } else {
+                    Token::Identifier(identifier)
+                }
+            }
```
Comment (kraenhansen) [inline src/lexer/mod.rs:620]: This got introduced since I have `rustfmt` running in my editor on save.

File: tests/parse_test.rs [hunk 2]
```diff
@@ -50,7 +50,8 @@ fn parse_mixin() {
             partial interface mixin Name {
                 readonly attribute unsigned short entry;
             };",
-    ).unwrap();
+    )
+    .unwrap();
```
Comment (kraenhansen) [inline tests/parse_test.rs:54]: This is also a change from `rustfmt`.

File: tests/parse_test.rs [hunk 3]
```diff
@@ -85,7 +86,8 @@ fn parse_integer_literals() {
              const unsigned long long max = 18446744073709551615;
              const long long min = -9223372036854775808;
          };",
-    ).unwrap();
+    )
+    .unwrap();
```
Comment (kraenhansen) [inline tests/parse_test.rs:90]: This too is a change from `rustfmt`.

**Comment by kraenhansen** [comment]
Looks like I'm missing something to make it pretty print correctly.

**Comment by kraenhansen** [comment]
I believe it's ready for a final review now.

**Comment by sgodwincs** [comment]
LGTM, thanks

**Pull request merged**